src/test/java/mate/academy/bookshop/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,255 @@
+                "Fiction description");
+        String requestBody = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(put("/categories/{id}", id)
+                        .content(requestBody)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        CategoryResponseDto actual = objectMapper.readValue(result
+                        .getResponse().getContentAsString(),
+                CategoryResponseDto.class);
+        CategoryResponseDto expected = new CategoryResponseDto(1L, "Fiction",
+                "Fiction description");
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @DisplayName("Delete existing category")
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    void delete_ExistingCategory_DeletesCategory() throws Exception {
+        Long id = 1L;
+
+        mockMvc.perform(delete("/categories/{id}", id)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNoContent());
+
+        MvcResult result = mockMvc.perform(get("/categories/{id}", id)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound())
+                .andReturn();
+
+        String actual = result.getResolvedException().getMessage();
+        String expected = "Category not found with id: " + id;
+        Assertions.assertEquals(expected, actual);
+    }
+
+    private List<CategoryResponseDto> getExpectedCategories() {
+        return IntStream.rangeClosed(1, 5)
+                .mapToObj(i -> {
+                    CategoryResponseDto categoryDto = new CategoryResponseDto(Long.valueOf(i),
+                            "Category " + i, "Description for Category " + i);
+                    return categoryDto;
+                })
+                .collect(Collectors.toList());
+    }
+
+    private List<BookDtoWithoutCategoryIds> getBooksWithoutCategories() {
+        return IntStream.rangeClosed(1, 20)
+                .mapToObj(i -> {
+                    BookDtoWithoutCategoryIds bookDtoWithoutCategoryIds =
+                            new BookDtoWithoutCategoryIds();
+                    bookDtoWithoutCategoryIds.setTitle("Book Title " + i);
+                    bookDtoWithoutCategoryIds.setAuthor("Author " + i);
+                    bookDtoWithoutCategoryIds.setPrice(BigDecimal.valueOf(10.99 + i - 1)
+                            .setScale(2, RoundingMode.HALF_UP));
+                    bookDtoWithoutCategoryIds.setDescription("Description for Book " + i);
+                    bookDtoWithoutCategoryIds.setCoverImage("coverImage" + i + ".jpg");
+                    return bookDtoWithoutCategoryIds;
+                })
+                .toList();
+    }
+
```

Comment:
OPTIONAL: It would be great to substract all logic that created object for testing to separate class and make a lot of constants for strings and numbers that you use